[CS2113T-W13-3] CLIMods

SuibianP opened the pull request:
A one-stop-shop for all needs of NUS students when it comes to schedules!

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,30 +1,97 @@
+* and adding Tasks `addTask()`.
+
+The Class diagrams for the different Tasks:  
+<img src="https://github.com/AY2122S1-CS2113T-W13-3/tp/blob/master/docs/images/Task%20Inheritence.jpeg?raw=true" alt="TodoFactory Sequence Diagram" width="650"/>  
```
Is there a spelling error here?
![image](https://user-images.githubusercontent.com/69423287/139169844-d185f907-3507-4bb1-a514-eddb0b44fc63.png)

ngnigel99 replied:
Should the inheritance arrows be pointing upwards?
![image](https://user-images.githubusercontent.com/69423287/139170226-57709f7e-0640-4d17-978f-5ad58d2f546b.png)

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,30 +1,97 @@
+<img src="https://github.com/AY2122S1-CS2113T-W13-3/tp/blob/master/docs/images/Task%20Enums.jpeg?raw=true" alt="TodoFactory Sequence Diagram" width="600"/>  
+
+The creation of tasks with the td, deadline r event cmmands are done using their respecive Task Factories  
+<img src="https://github.com/AY2122S1-CS2113T-W13-3/tp/blob/master/docs/images/TodoFactory%20Sequence%20Diagram.JPG?raw=true" alt="TodoFactory Sequence Diagram" width="500"/>  
```
Should these entities have the notation starting with `:`, e.g.  `:TodoFactory` instead?
![image](https://user-images.githubusercontent.com/69423287/139170902-3d479d90-2665-4da6-b513-df982a02d2d7.png)

ngnigel99 commented on docs/DeveloperGuide.md:
```diff
@@ -1,30 +1,97 @@
+Fixed values such as priority and recurrence are stored as an enum to ensure standardisation and that there are no invalid values being stored. The TaskTypes are also stored as an enum so that we can easily get the taskType when we have the task (and by extension the name of the task) for anything to do with parsing.  
+<img src="https://github.com/AY2122S1-CS2113T-W13-3/tp/blob/master/docs/images/Task%20Enums.jpeg?raw=true" alt="TodoFactory Sequence Diagram" width="600"/>  
+
+The creation of tasks with the td, deadline r event cmmands are done using their respecive Task Factories  
```
Possible spelling error here

ngnigel99 reviewed (commented):
(DG review during tutorial)
I like the elaboration in between diagrams, however, there are some minor changes that could be fixed to improve the readability

APZH committed:
Merge pull request #162 from APZH/branch-ExceptionsJavaDocs

add JavaDocs for exception classes

SeanRobertDH committed:
Update FileCreator getTaskFileName() to return absolute path

SeanRobertDH committed:
Merge pull request #163 from SeanRobertDH/master

Update FileCreator getTaskFileName() to return absolute path

SeanRobertDH committed:
Update FileCreator.java

Made save file plural

SeanRobertDH committed:
Update UserGuide.md

Add disclaimer to storage

SeanRobertDH committed:
Merge pull request #164 from AY2122S1-CS2113T-W13-3/branch-UGStorage

Update UserGuide.md

SeanRobertDH committed:
Update IOException Messages to show full file path.

SeanRobertDH committed:
Merge pull request #165 from SeanRobertDH/master

Update IOException Messages to show full file path.

APZH committed:
add PPP template for myself (APZH)

SeanRobertDH committed:
Update UserGuide.md

Make Todo, Deadline and Event usage reflect updated program

SeanRobertDH committed:
Merge pull request #166 from AY2122S1-CS2113T-W13-3/branch-DGTaskUsage

Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

APZH committed:
update the PPP for APZH

APZH committed:
minor tweak to APZH PPP

APZH committed:
Merge pull request #167 from APZH/branch-APZH_PPP

Branch apzh ppp

SeanRobertDH committed:
Update UserGuide.md

APZH committed:
fix APZH PPP image issues

SeanRobertDH committed:
Update UserGuide.md

APZH committed:
update portfolio link to point to APZH PPP

APZH committed:
Merge pull request #168 from APZH/master

Fix PPP image issues for APZH

APZH committed:
fix AboutUs portfolio link for APZH

APZH committed:
Merge pull request #169 from APZH/master

fix AboutUs portfolio link for APZH

SeanRobertDH committed:
Add default html file

SeanRobertDH committed:
Merge pull request #170 from SeanRobertDH/master

Add default html file

SeanRobertDH committed:
Update default.html

Test links

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Add css style

SeanRobertDH committed:
Merge pull request #171 from SeanRobertDH/master

Add css style

SeanRobertDH committed:
Update style.scss

SeanRobertDH committed:
Update style.scss

SeanRobertDH committed:
Add scroll to top button

SeanRobertDH committed:
Merge branch 'master' into master

SeanRobertDH committed:
Merge pull request #172 from SeanRobertDH/master

Add scroll to top button

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update default.html

Revert to js that workerd

SeanRobertDH committed:
Update style.scss

SeanRobertDH committed:
Update style.scss

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update AboutUs.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update edit command to check for invalid flags

SeanRobertDH committed:
Merge branch 'master' of https://github.com/AY2122S1-CS2113T-W13-3/tp

SeanRobertDH committed:
Merge pull request #173 from SeanRobertDH/master

Update edit command to check for invalid flags

SeanRobertDH committed:
Update EditCommand.java

SeanRobertDH committed:
Update UserGuide.md

Update edit command

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

Add delete command usage

APZH committed:
fix PPP link to APZH portfolio

APZH committed:
add missing line separators in the 'list' and 'sort' command expected output

SeanRobertDH committed:
Fix bug where Todo recurrence is not set if there is no date

APZH committed:
updated sequence diagram for filtering tasklist to include listcommand object

APZH committed:
add description to "Filtering Tasklist" section of the DG

APZH committed:
add user storeis to DG

APZH committed:
update PPP to reflect user stories contribution

APZH committed:
fix application naming bugs

APZH committed:
add instructions for manual testing for "Launching and Exit" to the DG

APZH committed:
add manual testing instructions for "Listing the Tasklist", "Filtering the Tasklist", "Sorting the Tasklist", and "Viewing the recurrence of a task"

APZH committed:
update PPP to reflect additional contributions made for APZH

APZH committed:
Merge pull request #174 from APZH/updateDocumentation-UG_DG_PPP

Update documentation ug dg ppp

APZH committed:
fix broken PPP link in AboutUs.md

APZH committed:
Merge pull request #175 from APZH/updateDocumentation-UG_DG_PPP

fix broken PPP link in AboutUs.md

APZH committed:
fix image not displaying bug in PPP

APZH committed:
Merge pull request #176 from APZH/updateDocumentation-UG_DG_PPP

fix image not displaying bug in PPP

SeanRobertDH committed:
Merge pull request #177 from SeanRobertDH/master

Fix bug where Todo recurrence is not set if there is no date

SeanRobertDH committed:
Change name from Duke to SchedUrMods and add object diagram for TaskManager

SeanRobertDH committed:
Merge pull request #178 from SeanRobertDH/master

Change name from Duke to SchedUrMods and add object diagram for TaskM…

SeanRobertDH committed:
Rename package and main class

SeanRobertDH committed:
replace occurences of 'duke' with 'schedUrMods'

SeanRobertDH committed:
Merge remote-tracking branch 'tp/master'

SeanRobertDH committed:
Merge pull request #180 from SeanRobertDH/master

Rename package and main class

SuibianP committed:
Implement upcoming command

List all tasks in chrono order in the upcoming week

The period can be made configurable

SuibianP committed:
Merge branch 'master' into feature-upcoming-command

APZH committed:
Merge pull request #181 from SuibianP/feature-upcoming-command

Implement upcoming command

SuibianP committed:
Add has link indicator to Lesson description

SuibianP committed:
Merge remote-tracking branch 'team/master' into feature-lesson-link-indicator

# Conflicts:
#	src/main/java/seedu/tp/task/type/Lesson.java

SuibianP committed:
Add DG and JavaDoc for nusmods package

APZH committed:
improved clarity of PPP and added PRs for DG and UG updates

APZH committed:
Merge pull request #184 from APZH/apzh-PPP

improved clarity of PPP and added PRs for DG and UG updates for APZH

APZH committed:
fix minor bug with PPP

APZH committed:
Merge pull request #185 from APZH/apzh-PPP

fix minor bug with PPP

SuibianP committed:
Merge remote-tracking branch 'team/master' into docs-nusmods

APZH committed:
Merge pull request #183 from SuibianP/feature-lesson-link-indicator

Add has link indicator to Lesson description

SuibianP committed:
Merge remote-tracking branch 'team/master' into docs-nusmods

SuibianP committed:
Add UG parts for module and browse commands

APZH committed:
Merge pull request #186 from SuibianP/docs-nusmods

Add UG, DG and JavaDoc for Mods parts

APZH committed:
update PPP for APZH to remove unecessary information

APZH committed:
add page break before DG optional extracts

SeanRobertDH committed:
Add uml diagrams and refactor TaskFactories

SuibianP committed:
Amend command summary for module and browse

APZH committed:
Merge pull request #188 from APZH/master

update PPP quality

SeanRobertDH committed:
Merge pull request #187 from SeanRobertDH/master

Add uml diagrams and refactor TaskFactories

SeanRobertDH committed:
Update default.html

Don't print button on convert to pdf

SuibianP committed:
Print friendly message if no upcoming tasks

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update default.html

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Move uml diagrams into individual file

SeanRobertDH committed:
Merge pull request #190 from SuibianP/feature-no-upcoming

Print friendly message if no upcoming tasks

SeanRobertDH committed:
Merge pull request #191 from SeanRobertDH/master

Move uml diagrams into individual file

SeanRobertDH committed:
Merge pull request #189 from SuibianP/docs-nusmods

Amend command summary for module and browse

SuibianP committed:
Add priority indicator for lesson description

SuibianP committed:
Check module-class pair exists and error if not

SeanRobertDH committed:
Change diagram and rename variables in Lesson

SeanRobertDH committed:
Merge pull request #192 from SuibianP/feature-lesson-priority-printout

Add priority indicator for lesson description

SeanRobertDH committed:
Merge pull request #193 from SuibianP/fix-check-module-exist

Check module-class pair exists and error if not

SeanRobertDH committed:
Merge pull request #194 from SeanRobertDH/master

Change diagram and rename variables in Lesson

SeanRobertDH committed:
Fix checkstyle issue

SeanRobertDH committed:
Update DeveloperGuide.md

APZH committed:
tweak task description of Lesson objects + minor quality adjustments to user guide

APZH committed:
Merge pull request #195 from APZH/master

tweak task description of Lesson objects + minor quality adjustments to UG

APZH committed:
add page break between the different diagrams to neaten the PPP

APZH committed:
remove 1 page break

SuibianP committed:
Add Hu Jialun PPP

APZH committed:
Merge pull request #196 from APZH/master

fix conversion to PDF format for PPP

APZH committed:
remove space between author tags and methods in Command Parser class

APZH committed:
Merge pull request #197 from APZH/master

remove space between author tags and methods in Command Parser class

SeanRobertDH committed:
Update DeveloperGuide.md

Add logger, storage and TaskManager components

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Add severity to logger message

SeanRobertDH committed:
Merge pull request #198 from SeanRobertDH/master

Add severity to logger message

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Used constant naming convention

SeanRobertDH committed:
add Task Factory Object Diagram

SeanRobertDH committed:
Merge pull request #199 from SeanRobertDH/master

add task factory diagram

SeanRobertDH committed:
Add TaskFactory to Implementation

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Give edit command usage if no flags are provided

SeanRobertDH committed:
Check when user puts the description flag but no description

SeanRobertDH committed:
Update DeveloperGuide.md

Add testing methodology for Task, edit and delete commands

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Fix logger class, make FileCreator check that file wasn't already in use

SeanRobertDH committed:
Update DeveloperGuide.md

Add granular architecture diagram

SeanRobertDH committed:
Merge pull request #200 from SeanRobertDH/master

Last minute bug fixes

SeanRobertDH committed:
delete images

SeanRobertDH committed:
Merge pull request #201 from SeanRobertDH/master

delete images

SeanRobertDH committed:
readd images

SeanRobertDH committed:
Merge pull request #202 from SeanRobertDH/master

readd images

SeanRobertDH committed:
Add personal PPP

SeanRobertDH committed:
Merge pull request #203 from SeanRobertDH/master

Add personal PPP

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update seanrobertdh.md

SuibianP committed:
Rename Hu Jialun PPP to GitHub username

SuibianP committed:
Amend PPP to add more contributions

SuibianP committed:
Change to precise wordings

SuibianP committed:
Merge pull request #204 from SuibianP/doc-hujialun-ppp

Add Hu Jialun PPP

SuibianP committed:
Add nusmods data pulling

SuibianP committed:
Add class diagram for NusModsParser and Semester

SuibianP committed:
Merge pull request #205 from SuibianP/docs-nusmods

Add class diagram for NusModsParser and Semester

Xuefei2001 committed:
no message

Xuefei2001 committed:
Resolve merge conflicts

Xuefei2001 committed:
Resolve merge conflict for UG

SeanRobertDH committed:
Update the user guide

SeanRobertDH committed:
Merge pull request #206 from SeanRobertDH/master

Update the user guide

SuibianP committed:
Add friendly message if the browsed task is not a lesson

SuibianP committed:
Merge pull request #207 from SuibianP/feature-task-not-lesson

Add friendly message if the browsed task is not a lesson

SuibianP committed:
Link AboutUs.md portfolio to PPP

SuibianP committed:
Amend instructions for zip releases

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SuibianP committed:
Add upcoming command in user guide

SuibianP committed:
Add dead link test in PPP

SeanRobertDH committed:
Update UserGuide.md

program may not work on higher versions of java. should strictly be java 11

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

Xuefei2001 committed:
implement diary

Xuefei2001 committed:
no message

Xuefei2001 committed:
no message

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

Xuefei2001 committed:
Update UG

Xuefei2001 committed:
Merge branch 'master' into master

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update style.scss

Se if @media tag works to hide arrow when printing

SuibianP committed:
Break long code lines to ensure print quality

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Merge branch 'master' into doc-zip

SeanRobertDH committed:
Merge pull request #208 from SuibianP/doc-zip

Amend instructions for zip releases

SeanRobertDH committed:
Update UserGuide.md

SuibianP committed:
Add upcoming command in developer guide

SeanRobertDH committed:
Merge pull request #209 from SuibianP/doc-ugdg-upcoming

Add upcoming command in user guide and developer guide

SeanRobertDH committed:
Add recurrence to edit help

SeanRobertDH committed:
add 'none' recurrence as option

SeanRobertDH committed:
replace getHappenTime() with getListDate()

SeanRobertDH committed:
Merge pull request #211 from SeanRobertDH/master

Minor code refactors

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update apzh.md

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update suibianp.md

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update UserGuide.md

Xuefei2001 committed:
Update DG

SuibianP committed:
Add Xuefei PPP template

Xuefei2001 committed:
Merge branch 'master' of https://github.com/Xuefei2001/tp

SuibianP committed:
Merge remote-tracking branch 'team/master'

Xuefei2001 committed:
Delete reminderinformation

Xuefei2001 committed:
Merge branch 'master' into master

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update seanrobertdh.md

SeanRobertDH committed:
Update seanrobertdh.md

Xuefei2001 committed:
Add reminder to proposed implementation in DG

SeanRobertDH committed:
Update apzh.md

SuibianP committed:
Merge remote-tracking branch 'team/master'

SeanRobertDH committed:
Update UserGuide.md

SuibianP committed:
Add reminder to TOC

SeanRobertDH committed:
Update DeveloperGuide.md

Xuefei2001 committed:
Update author

Xuefei2001 committed:
Merge branch 'master' of https://github.com/Xuefei2001/tp

SeanRobertDH committed:
Update DeveloperGuide.md

SeanRobertDH committed:
Update UserGuide.md

SeanRobertDH committed:
Update UserGuide.md

Xuefei2001 committed:
Add PPP

Xuefei2001 committed:
Merge branch 'master' into master

Xuefei2001 committed:
Merge pull request #210 from Xuefei2001/master

Update UG an reminder

SuibianP committed:
Revert "Update UG an reminder"

SeanRobertDH committed:
Merge pull request #212 from AY2122S1-CS2113T-W13-3/revert-210-master

Revert "Update UG an reminder"
